pkg/server/plugin/datastore/sql/sql_test.go

```diff
@@ -815,52 +816,121 @@ func (s *PluginSuite) TestFetchAttestedNodesWithPagination() {
-		SpiffeId:         node.SpiffeId,
-		CertSerialNumber: userial,
-		CertNotAfter:     uexpires,
-	})
-	s.RequireGRPCStatus(err, codes.NotFound, _notFoundErrMsg)
-
-	_, err = s.ds.CreateAttestedNode(ctx, &datastore.CreateAttestedNodeRequest{Node: node})
-	s.Require().NoError(err)
-
-	uresp, err := s.ds.UpdateAttestedNode(ctx, &datastore.UpdateAttestedNodeRequest{
-		SpiffeId:         node.SpiffeId,
-		CertSerialNumber: userial,
-		CertNotAfter:     uexpires,
-	})
-	s.Require().NoError(err)
-
-	unode := uresp.Node
-	s.Require().NotNil(unode)
+	fullUpdatedNode := &common.AttestedNode{
+		SpiffeId:            node.SpiffeId,
+		AttestationDataType: "aws-tag",
+		CertSerialNumber:    userial,
+		CertNotAfter:        uexpires,
+		NewCertNotAfter:     unewExpires,
+		NewCertSerialNumber: unewSerial,
+	}
 
-	s.Equal(node.SpiffeId, unode.SpiffeId)
-	s.Equal(node.AttestationDataType, unode.AttestationDataType)
-	s.Equal(userial, unode.CertSerialNumber)
-	s.Equal(uexpires, unode.CertNotAfter)
+	for _, tt := range []struct {
+		name           string
+		updateReq      *datastore.UpdateAttestedNodeRequest
+		existentNode   *common.AttestedNode
+		expUpdatedNode *common.AttestedNode
+		expErr         error
+	}{
+		{
+			name: "update non-existing attested node",
+			updateReq: &datastore.UpdateAttestedNodeRequest{
+				SpiffeId:         node.SpiffeId,
+				CertSerialNumber: userial,
+				CertNotAfter:     uexpires,
+			},
+			expErr: status.Error(codes.NotFound, _notFoundErrMsg),
+		},
+		{
+			name: "update attested node with all false mask",
+			updateReq: &datastore.UpdateAttestedNodeRequest{
+				SpiffeId:            node.SpiffeId,
+				CertSerialNumber:    userial,
+				CertNotAfter:        uexpires,
+				NewCertNotAfter:     unewExpires,
+				NewCertSerialNumber: unewSerial,
+				InputMask:           &common.AttestedNodeMask{},
+			},
+			existentNode:   node,
+			expUpdatedNode: node,
+		},
+		{
+			name: "update attested node with mask set only some fields: 'CertSerialNumber', 'NewCertNotAfter'",
+			updateReq: &datastore.UpdateAttestedNodeRequest{
+				SpiffeId:            node.SpiffeId,
+				CertSerialNumber:    userial,
+				CertNotAfter:        uexpires,
+				NewCertNotAfter:     unewExpires,
+				NewCertSerialNumber: unewSerial,
+				InputMask: &common.AttestedNodeMask{
+					CertSerialNumber: true,
+					NewCertNotAfter:  true,
+				},
+			},
+			existentNode: node,
+			expUpdatedNode: func() *common.AttestedNode {
+				n := &common.AttestedNode{}
+				*n = *node
+				n.CertSerialNumber = userial
+				n.NewCertNotAfter = unewExpires
+				return n
+			}(),
+		},
+		{
+			name: "update attested node with nil mask",
+			updateReq: &datastore.UpdateAttestedNodeRequest{
+				SpiffeId:            node.SpiffeId,
+				CertSerialNumber:    userial,
+				CertNotAfter:        uexpires,
+				NewCertNotAfter:     unewExpires,
+				NewCertSerialNumber: unewSerial,
+			},
+			existentNode:   node,
+			expUpdatedNode: fullUpdatedNode,
```

Comment:
full is only used here can you initialize that var here? 
it will make test case easier to read